### PR TITLE
POLIO-858: Polio / Budget: migrate repeat button to steps

### PR DESCRIPTION
**Pull request by beygorghor**
Whimsical [link](https://whimsical.com/move-repeat-step-button-8JFFU8XomPLPhKfoxp9fUu)

Moving repeat step button form the top of the page to the correct row of step in step table

Related JIRA tickets : POLIO-858

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [x] New translations have been added or updated if new string have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- removed the option to display a repeat action from `AddStepButton` and add `RepeatStepIconProps`
- pre-filter repeat steps in `BudgetDetailsInfos` and pass it to steps table
- Create an action cell for steps
- display repeat button on steps with a repeat action possible

## How to test

Open an existing budget for a specific campaign and country.
Proceed to steps.
You should see the loop icon (mainly on the last step)
Click on it and re submit the step

## Print screen / video

https://user-images.githubusercontent.com/12494624/223716910-5118356e-7ade-4721-97ce-4bfd901fe4d8.mov

## Notes

I'm wondering if we should not display that a step has been repeated in the table

### CODE REVIEW

File: plugins/polio/js/src/pages/Budget/CreateBudgetStep/AddStepButton.tsx [hunk 1]
```diff
@@ -1,7 +1,17 @@
 import React, { FunctionComponent } from 'react';
-import { Button, makeStyles, useMediaQuery, useTheme } from '@material-ui/core';
+// import { IconButton } from 'bluesquare-components';
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/CreateBudgetStep/AddStepButton.tsx:2]: you can delete this

File: plugins/polio/js/src/pages/Budget/hooks/config.tsx [hunk 2]
```diff
@@ -239,32 +247,39 @@ export const useBudgetDetailsColumns = (): Column[] => {
...
+        getRowColor,
+        classes.paragraph,
+        params,
+        budgetDetails,
+        repeatTransitions,
+    ]);
 };
 
-type Params = {
+type TablePorps = {
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/hooks/config.tsx:271]: TablePorps 😂

**Review by quang-le — CHANGES_REQUESTED** [review]
We get a strange UX when hiding steps: if we hide the last step, we can't repeat it anymore unless we show hidden steps. I think this will confuse the users. We should probably sit down and spec this:
https://user-images.githubusercontent.com/38907762/223746288-48434216-584f-49a8-a4cc-22b2ba690e67.mov

**Comment by beygorghor** [comment]
> We get a strange UX when hiding steps: if we hide the last step, we can't repeat it anymore unless we show hidden steps. I think this will confuse the users. We should probably sit down and spec this: https://user-images.githubusercontent.com/38907762/223746288-48434216-584f-49a8-a4cc-22b2ba690e67.mov

@quang-le maybe a warning saying you have hidden steps that can be repeated ??

**Review by quang-le — APPROVED** [review]

**Pull request merged**